src/MarkdownTextInput.web.tsx

```diff
@@ -1,22 +1,438 @@
-import {TextInput} from 'react-native';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type {
+  TextInput,
+  TextInputSubmitEditingEventData,
+  TextStyle,
+  NativeSyntheticEvent,
+  TextInputSelectionChangeEventData,
+  TextInputProps,
+  TextInputKeyPressEventData,
+  TextInputFocusEventData,
+} from 'react-native';
+import React, {useEffect, useRef, useCallback, useMemo} from 'react';
+import type {CSSProperties, MutableRefObject, ReactEventHandler, FocusEventHandler, MouseEvent, KeyboardEvent, SyntheticEvent} from 'react';
+import {StyleSheet} from 'react-native';
+import * as ParseUtils from './web/parserUtils';
+import * as CursorUtils from './web/cursorUtils';
+import * as StyleUtils from './styleUtils';
+import type * as MarkdownTextInputDecoratorViewNativeComponent from './MarkdownTextInputDecoratorViewNativeComponent';
+import './web/MarkdownTextInput.css';
+import MarkdownInputHistory from './web/historyClass';
 
-import React from 'react';
-import type {TextInputProps} from 'react-native';
+// TODO: remove require
+require('../parser/react-native-live-markdown-parser.js');
+
+let createReactDOMStyle: (style: any) => any;
+try {
+  createReactDOMStyle =
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    require('react-native-web/dist/exports/StyleSheet/compiler/createReactDOMStyle').default;
+} catch (e) {
+  throw new Error('[react-native-live-markdown] Function `createReactDOMStyle` from react-native-web not found. Please make sure that you are using React Native Web 0.18 or newer.');
+}
+
+let preprocessStyle: (style: any) => any;
+try {
+  preprocessStyle =
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    require('react-native-web/dist/exports/StyleSheet/preprocess').default;
+} catch (e) {
+  throw new Error('[react-native-live-markdown] Function `preprocessStyle` from react-native-web not found.');
+}
+
+type MarkdownStyle = MarkdownTextInputDecoratorViewNativeComponent.MarkdownStyle;
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
 interface MarkdownTextInputProps extends TextInputProps {
-  // nothing here
+  markdownStyle?: MarkdownStyle;
+  onClick?: (e: MouseEvent<HTMLDivElement>) => void;
+  dir?: string;
+}
+
+interface MarkdownNativeEvent extends Event {
+  inputType: string;
+}
+
+type Selection = {
+  start: number;
+  end: number;
+};
+
+let focusTimeout: NodeJS.Timeout | null = null;
+
+// Removes one '\n' from the end of the string that were added by contentEditable div
+const normalizeValue = (value: string) => {
+  return value.replace(/\n$/, '');
+};
```

Comment:
Can we convert all arrow functions into function declarations? 
```suggestion
function normalizeValue(value: string) {
  return value.replace(/\n$/, '');
}
```